File: scala/src/main/scala/model/Order.scala
package model

import java.time.LocalDateTime

class Order {

  private var itemList: List[Item] = Nil
  private var customerName: String = ""
  private var contact: String = ""
  private var shippingAddres: String = ""
  private var grandTotal = .0
  private var createDate: LocalDateTime = LocalDateTime.now()

  def constructor(newItemList: List[Item], newCustomerName: String, newContact: String, newShippingAddres: String, newGrandTotal: Double, newCreateDate: LocalDateTime): Unit = {
    itemList = newItemList
    customerName = newCustomerName
    contact = newContact
    shippingAddres = newShippingAddres
    grandTotal = newGrandTotal
    createDate = newCreateDate
  }

  def addItem(newItem: Item): Unit = {
    itemList = itemList :+ newItem
  }
  def setItemList(newItemList: List[Item]): Unit = {
     itemList = newItemList
  }
  def getItemList: List[Item] = itemList

  def setCustomerName(newCustomerName: String): Unit = {
    customerName = newCustomerName
  }
  def getCustumerName: String = customerName

  def setContact(newContact: String): Unit = {
    contact = newContact
  }
  def getContact: String = contact

  def setShippingAddres(newShippingAddres: String): Unit = {
    shippingAddres = newShippingAddres
  }
  def getShippingAddres: String = shippingAddres

  def setGrandTotal(newTotal: Double): Unit = {
    grandTotal = newTotal
  }
  def getGrandTotal: Double = grandTotal

  def setCreateDate(newCreateDate: LocalDateTime): Unit = {
    createDate = newCreateDate
  }
  def getCreateDate: LocalDateTime = createDate
}

File: java/src/main/model/Order.java
package main.model;

import java.time.LocalDateTime;
import java.util.List;

public class Order {

    private List<Item> itemList;
    private String customerName;
    private String contact;
    private String shippingAddres;
    private double grandTotal;
    private LocalDateTime createDate;

    public Order(List<Item> itemList, String customerName, String contact, String shippingAddres, double grandTotal, LocalDateTime createDate) {
        this.itemList = itemList;
        this.customerName = customerName;
        this.contact = contact;
        this.shippingAddres = shippingAddres;
        this.grandTotal = grandTotal;
        this.createDate = createDate;
    }

    public Order(List<Item> itemList, String customerName, String contact, String shippingAddres, double grandTotal) {
        this.itemList = itemList;
        this.customerName = customerName;
        this.contact = contact;
        this.shippingAddres = shippingAddres;
        this.grandTotal = grandTotal;
        this.createDate = LocalDateTime.now();
    }

    public List<Item> getItemList() {
        return itemList;
    }

    public void setItemList(List<Item> itemList) {
        this.itemList = itemList;
    }

    public String getCustomerName() {
        return customerName;
    }

    public void setCustomerName(String customerName) {
        this.customerName = customerName;
    }

    public String getContact() {
        return contact;
    }

    public void setContact(String contact) {
        this.contact = contact;
    }

    public String getShippingAddres() {
        return shippingAddres;
    }

    public void setShippingAddres(String shippingAddres) {
        this.shippingAddres = shippingAddres;
    }

    public double getGrandTotal() {
        return grandTotal;
    }

    public void setGrandTotal(double grandTotal) {
        this.grandTotal = grandTotal;
    }

    public LocalDateTime getCreateDate() {
        return createDate;
    }

    public void setCreateDate(LocalDateTime createDate) {
        this.createDate = createDate;
    }
}
